### PR TITLE
35229 navbar

### PR DESCRIPTION
**Pull request by Sam152**

### CODE REVIEW

File: agov/themes/agov/agov_whitlam/agov_whitlam.theme [hunk 1]
```diff
@@ -58,3 +58,37 @@ function agov_whitlam_preprocess_breadcrumb(&$vars) {
       ->applyTo($vars);
   }
 }
+
+/**
+ * Implements hook_preprocess_block().
+ */
+function agov_whitlam_preprocess_block(&$vars) {
```
Comment (benjy) [inline agov/themes/agov/agov_whitlam/agov_whitlam.theme]: This and the one below, feels like there would be a more specific them hook?

File: agov/themes/agov/agov_whitlam/sass/components/primary-navigation/primary-navigation.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+(function ($) {
+  Drupal.behaviors.agov_whitlam_primary_navigation = {
+    attach: function (context, settings) {
+      $('.primary-navigation', context).once().each(function() {
```
Comment (benjy) [inline agov/themes/agov/agov_whitlam/sass/components/primary-navigation/primary-navigation.js]: Doesn't once() take a function as a param? Why do we use each()?

**Comment by benjy** [comment]
Code looks good but the build failed so I couldn't review the end product.

File: agov/themes/agov/agov_whitlam/agov_whitlam.theme (hunk 1 above)
Comment (Sam152) [inline agov/themes/agov/agov_whitlam/agov_whitlam.theme]: I think it's better to make this a broader hook so we can continue to add elements to the array when we need to add classes to blocks in the future.

File: agov/themes/agov/agov_whitlam/sass/components/primary-navigation/primary-navigation.js (hunk 2 above)
Comment (Sam152) [inline agov/themes/agov/agov_whitlam/sass/components/primary-navigation/primary-navigation.js]: New API for the version of jQuery. The old once looped over every element, but now it's more explicit.

**Comment by Sam152** [comment]
Will try get this build green.

File: agov/themes/agov/agov_whitlam/agov_whitlam.theme (hunk 1 above)
Comment (benjy) [inline agov/themes/agov/agov_whitlam/agov_whitlam.theme]: Yeah, may as well put it in a block module :/

Comment (Sam152) [inline agov/themes/agov/agov_whitlam/agov_whitlam.theme]: How is it like that at all? It's essentially whether adding classes should go in one function or multiple functions. This is more dry and I'd suggest we move the other block hook to be in here as well if possible.

**Comment by Sam152** [comment]
test this please

**Comment by benjy** [comment]
test this please

**Comment by nickschuch** [comment]
test this please

**Comment by benjy** [comment]
I think we need a before/after screenshot of the 3 theme vs now. Initial thoughts are the Menu text is all wrong, and we've lost the icons.

**Comment by Sam152** [comment]
8 on the left, 3 on the right.

screen shot 2015-10-14 at 1 51 15 pm
screen shot 2015-10-14 at 1 50 37 pm
screen shot 2015-10-14 at 1 50 21 pm

File: agov/themes/agov/agov_whitlam/agov_whitlam.theme [hunk 3]
```diff
@@ -58,3 +44,38 @@ function agov_whitlam_preprocess_breadcrumb(&$vars) {
...
+    $vars['title_attributes']['class'] = $title_classes[$vars['plugin_id']];
+  }
+}
+
+/**
+ * Implements hook_preprocess_menu().
+ */
+function agov_whitlam_preprocess_menu(&$vars) {
+  // @todo, is there better context?
+  if ($vars['theme_hook_original'] === 'menu__main') {
```
Comment (benjy) [inline agov/themes/agov/agov_whitlam/agov_whitlam.theme]: I think this logic should be moved to hook_theme_suggestions_HOOK_alter

Comment (Sam152) [inline agov/themes/agov/agov_whitlam/agov_whitlam.theme]: Will look at during a follow up right now.

**Pull request merged**